configure.ac

```
@@ -1134,16 +1134,120 @@ char** (*fn)(void* const*, BACKTRACE_RETURN_TYPE) = backtrace_symbols;
+      ;;
+   *)
+      AC_MSG_ERROR([bad value '$enable_backtrace' for --enable-backtrace])
+      ;;
+esac
+if test "x$enable_backtrace" != xno; then
+   AC_DEFINE([HAVE_BACKTRACE_SCREEN], [1], [Define if the backtrace feature is enabled])
+fi
+AM_CONDITIONAL([HAVE_BACKTRACE_SCREEN], [test x"$enable_backtrace" != xno])
+
+if test "x$enable_backtrace" = xunwind-ptrace -a "x$my_htop_platform" = xlinux; then
+   AC_MSG_CHECKING([if libunwind has unw_get_elf_filename])
+   AC_LINK_IFELSE(
+      [AC_LANG_PROGRAM(
+         [[/* unw_get_elf_filename might exist as a macro */
+         #include <libunwind-ptrace.h>]],
+         [[static unw_cursor_t cursor;
+         /* First argument must be non-null */
+         unw_get_elf_filename(&cursor, (char*)0, (size_t)0, (unw_word_t*)0);]]
+      )],
+      [AC_DEFINE([HAVE_LIBUNWIND_ELF_FILENAME], [1], [Define if libunwind has unw_get_elf_filename])
+      AC_MSG_RESULT(yes)],
+      [AC_MSG_RESULT(no)]
+   )
+fi
+
+AC_ARG_ENABLE([demangling],
+              [AS_HELP_STRING([--enable-demangling],
+                              [enable demangling support for backtraces @<:@default=check@:>@])],
+              [],
+              [enable_demangling=check])
+have_libiberty=no
+case $enable_demangling in
+   check|yes|libiberty)
+      AC_CHECK_LIB([iberty], [cplus_demangle], [
+         have_libiberty=yes
+         AC_DEFINE([HAVE_LIBIBERTY], [1], [Define to 1 if libiberty is found.])
+         LIBS="$LIBS -liberty"
+         ], [
+         if test "x$enable_demangling" = xlibiberty; then
+            AC_MSG_ERROR([--enable-demangling specified but libiberty not found])
+         fi
+      ])
+
+      if test "x$have_libiberty" = xyes; then
+         if ! htop_search_header_dir libiberty/demangle.h "/usr/include/libiberty"; then
+            have_libiberty=no
+            if test "x$enable_demangling" = xlibiberty; then
+               AC_MSG_ERROR([--enable-demangling specified but <demangle.h> not found])
+            fi
+         fi
+      fi
+
+      if test "x$have_libiberty" = xyes; then
+         enable_demangling=libiberty
+      fi
+      ;;
+esac
+
+case $enable_demangling in
+   libiberty)
+      AC_DEFINE([HAVE_DEMANGLING], [1], [Define to 1 if the demangling is supported.])
+      ;;
+   check)
+      enable_demangling=no
+      ;;
+   yes)
+      AC_MSG_ERROR([cannot find any library for the backend supported by --enable-demangling])
+      ;;
+   no)
+      ;;
+   *)
+      AC_MSG_ERROR([bad value '$enable_demangling' for --enable-demangling])
+      ;;
+esac
+
+if test "x$enable_backtrace" != xno && test "x$enable_demangling" = xno; then
+   AC_MSG_WARN([The backtrace feature is enabled, but name demangling is not.])
+fi
+
+AC_ARG_ENABLE([hwloc],
+              [AS_HELP_STRING([--enable-hwloc],
+              [enable hwloc support for CPU affinity; disables affinity support; requires libhwloc @<:@default=no@:>@])],
+              [],
+              [enable_hwloc=no])
```

This touches an unrelated check.
```suggestion
AC_ARG_ENABLE(
   [hwloc],
   [AS_HELP_STRING(
      [--enable-hwloc],
      [enable hwloc support for CPU affinity; disables affinity support; requires libhwloc @<:@default=no@:>@]
   )],
   [],
   [enable_hwloc=no]
)
```